assimilation_code/modules/utilities/parse_args_mod.f90

```diff
@@ -46,16 +46,49 @@
 
 module parse_args_mod
 
-use utilities_mod, only : error_handler, E_ERR
+use types_mod,     only : r8, i8, i4, MISSING_R8
+use utilities_mod, only : error_handler, E_ERR, find_textfile_dims,    &
+                          file_exist, open_file, close_file, to_upper, &
+                          string_to_real, string_to_integer
 
 implicit none
 private
 
-public :: get_args_from_string, &
+public :: get_args_from_string,           &
           get_name_val_pairs_from_string, &
-          get_next_arg
-
-character(len=*), parameter :: source = 'parse_args_mod.f90'
+          get_next_arg,                   &
+          csv_get_obs_num,                &
+          csv_get_field,                  &
+          csv_file_type,                  &
+          csv_field_exists,               &
+          csv_print_header,               & 
+          csv_get_field_index,            & 
+          csv_open,                       &
+          csv_close
+
+interface csv_get_field
+    module procedure csv_get_field_char
+    module procedure csv_get_field_int
+    module procedure csv_get_field_real
+end interface csv_get_field
+
+character(len=*), parameter :: source         = 'parse_args_mod.f90'
+character(len=*), parameter :: EMPTY_ENTRY    = '_EMPTY_'            ! Used to fill in missing data in the raw file
+integer,          parameter :: MAX_FIELDS_LEN = 15000
+integer,          parameter :: MAX_NUM_FIELDS = 1000
+
+! a csv file structure
+type csv_file_type
+   character(len=256) :: filename = '' 
```

Comment:
i'd mark this private so the contents can't be accessed by any code other than this module.  the calling code can pass the handle around, but it can't mess up the internals.  you may need to add some additional access functions (e.g. return number of rows) but that's a cleaner solution.